src/shader_recompiler/ir/passes/resource_tracking_pass.cpp

```diff
@@ -656,6 +656,32 @@ void PatchImageInstruction(IR::Block& block, IR::Inst& inst, Info& info, Descrip
     }
     ASSERT(image.GetType() != AmdGpu::ImageType::Invalid);
     const bool is_storage = IsImageStorageInstruction(inst);
+
+    // Patch image instruction if image is fmask.
+    if (image.IsFmask()) {
+        // Ignore fmask writes. TODO: handle dimension queries.
```

Comment:
I'm not sure if FMask will be written in shader (but not excluding such possibility), so will be better to make an assertion on `is_storage` there